api/apps/api/src/modules/geo-feature-tags/geo-feature-tags.service.ts

```diff
@@ -0,0 +1,53 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { GeoFeatureTag } from '@marxan-api/modules/geo-feature-tags/geo-feature-tag.api.entity';
+import { Repository } from 'typeorm';
+import { GeoFeature } from '@marxan-api/modules/geo-features/geo-feature.api.entity';
+import { Either, left, right } from 'fp-ts/lib/Either';
+
+export const featureNotFound = Symbol('feature not found');
+export const featureProjectNotRelated = Symbol(
+  'feature and project not related',
+);
+
+export class GeoFeatureTagsService {
+  constructor(
+    @InjectRepository(GeoFeatureTag)
+    private readonly geoFeatureTagsRepo: Repository<GeoFeatureTag>,
+    @InjectRepository(GeoFeature)
+    private readonly geoFeaturesRepo: Repository<GeoFeature>,
+  ) {}
+
+  async updateTagForFeature(
```

Comment:
tiny style suggestion: we do often provide functions that allow to set or update something, where specifically we don't care if this something was already set or not - in these cases, I recommend to use a `setOrUpdateSomething` kind of name to make this immediately clear when reading the source code.